packages/server-wallet/src/wallet/index.ts

```diff
@@ -162,95 +164,66 @@ export class Wallet implements WalletInterface {
+    return await this.store.getOrCreateSigningAddress();
   }
 
   async createChannel(args: CreateChannelParams): SingleChannelResult {
-    const {participants, appDefinition, appData, allocations} = args;
+    const {participants, appDefinition, appData, allocations, fundingStrategy} = args;
     const outcome: Outcome = deserializeAllocations(allocations);
 
-    const channelNonce = await Nonce.next(
-      this.knex,
-      participants.map(p => p.signingAddress)
-    );
-    const channelConstants: ChannelConstants = {
+    const channelNonce = await this.store.nextNonce(participants.map(p => p.signingAddress));
+    const constants: ChannelConstants = {
       channelNonce,
       participants: participants.map(convertToParticipant),
       chainId: '0x01',
       challengeDuration: 9001,
       appDefinition,
     };
 
-    const vars: SignedStateVarsWithHash[] = [];
-
-    // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-    const callback = async (tx: Transaction) => {
-      // TODO: How do we pick a signing address?
-      const signingAddress = (await SigningWallet.query(tx).first())?.address;
+    const {outgoing, channelResult} = await this.store.createChannel(
+      constants,
+      appData,
+      outcome,
+      fundingStrategy
+    );
 
-      const cols = {
-        ...channelConstants,
-        vars,
-        signingAddress,
-        chainServiceRequests: [],
-        fundingStrategy: args.fundingStrategy,
+    // todo: clean up the construction of the Objective
+    // todo: do not assume Direct funding
+    const outbox: Outgoing[] = outgoing.map(n => {
+      const params = n.notice.params as {
+        sender: string;
+        recipient: string;
+        data: {signedStates: SignedState[]};
       };
-
-      const channel = await Channel.query(tx)
-        .insert(cols)
-        .returning('*');
-
-      const {outgoing, channelResult} = await this.store.signState(
-        channel.channelId,
-        {
-          ...channelConstants,
-          turnNum: 0,
-          isFinal: false,
-          appData,
-          outcome,
-        },
-        tx
-      );
-
-      // todo: clean up the construction of the Objective
-      // todo: do not assume Direct funding
-      const outbox: Outgoing[] = outgoing.map(n => {
-        const params = n.notice.params as {
-          sender: string;
-          recipient: string;
-          data: {signedStates: SignedState[]};
-        };
-        return {
-          method: 'MessageQueued' as const,
-          params: {
-            ...n.notice.params,
-            data: {
-              objectives: [
-                {
-                  participants,
-                  type: 'CreateChannel',
-                  data: {
-                    signedState: params.data.signedStates[0],
-                    fundingStrategy: 'Direct',
-                  },
+      return {
+        method: 'MessageQueued' as const,
```

Comment:
I see `as const` sprinkled around a fair bit. Do you know what is it doing for us?

Reply:
No idea 🤷 I think @kerzhner has been adding these

Reply:
It narrows the type from `string` to a specific string type